src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/RazorAssemblyLoadContext.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+#if NET
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.Loader;
+using System.Threading;
+using Microsoft.CodeAnalysis.ExternalAccess.Razor;
+
+namespace Microsoft.CodeAnalysis.Remote.Razor;
+
+internal sealed class RazorAssemblyLoadContext : AssemblyLoadContext
+{
+    private readonly AssemblyLoadContext? _parent;
+    private readonly string _baseDirectory;
+
+    private Assembly? _razorCompilerAssembly;
+
+    private object _loaderLock = new();
```

Comment:
Are we using .NET 9 here? Can we use `System.Threading.Lock`?

Reply:
We're still targeting NET8, don't really want to upgrade our TF as part of this PR